lib/fastly_nsq/message_queue/producer.rb

```diff
@@ -1,32 +1,33 @@
+require 'forwardable'
+
 class InvalidParameterError < StandardError; end
 
 module MessageQueue
   class Producer
-    def initialize(topic:, ssl_context: nil)
-      @topic = topic
-      @ssl_context = SSLContext.new(ssl_context)
-    end
+    extend Forwardable
+    def_delegator :connection, :terminate
+    def_delegator :connection, :write
 
-    def connection
-      @producer ||= producer.new(params)
-    end
-
-    def terminate
-      @producer.terminate
+    def initialize(topic:, ssl_context: nil, &connector)
```

Comment:
Yes! This would be helpful :+1:

On Wed, May 11, 2016 at 10:42 AM, alieander <notifications@github.com>
wrote:

> In lib/fastly_nsq/message_queue/producer.rb
> <https://github.com/fastly/fastly_nsq/pull/29#discussion_r62892696>:
>
> >
> > -    def connection
> > -      @producer ||= producer.new(params)
> > -    end
> > -
> > -    def terminate
> > -      @producer.terminate
> > +    def initialize(topic:, ssl_context: nil, &connector)
>
> Might it help if the README demonstrated this new usage pattern?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/fastly/fastly_nsq/pull/29/files/f057cf5482820ecaa08fd2243200c638279d8ca0#r62892696>
>



-- 
Adarsh Pandit
External Consultant
+1 313 454 1515 (m)
@adarshp <https://twitter.com/adarshp>